libraries/AP_RangeFinder/AP_RangeFinder_BLPing.cpp

```diff
@@ -66,12 +38,67 @@ void AP_RangeFinder_BLPing::update(void)
 
 void AP_RangeFinder_BLPing::init_sensor()
 {
-    // request distance from sensor
-    send_message(BLDPIN_MSGID_DISTANCE_SIMPLE, nullptr, 0);
+    // Set message interval between pings in ms
+    uint16_t ping_interval = _sensor_rate_ms;
+    protocol.send_message(uart, PingProtocol::MessageId::SET_PING_INTERVAL, reinterpret_cast<uint8_t*>(&ping_interval), sizeof(ping_interval));
+
+    // Send a message requesting a continuous
+    uint16_t continuous_message = static_cast<uint16_t>(PingProtocol::MessageId::DISTANCE_SIMPLE);
+    protocol.send_message(uart, PingProtocol::MessageId::CONTINUOUS_START, reinterpret_cast<uint8_t*>(&continuous_message), sizeof(continuous_message));
 }
 
-// send message to sensor
-void AP_RangeFinder_BLPing::send_message(uint16_t msgid, const uint8_t *payload, uint16_t payload_len)
+// distance returned in reading_cm, signal_ok is set to true if sensor reports a strong signal
+bool AP_RangeFinder_BLPing::get_reading(uint16_t &reading_cm)
+{
+    if (uart == nullptr) {
+        return false;
+    }
+
+    struct {
+        float sum_cm = 0;
+        uint16_t count = 0;
+        float mean() { return sum_cm / count; };
+    } averageStruct;
+
+    // read any available lines from the lidar
+    int16_t nbytes = uart->available();
+    while (nbytes-- > 0) {
+        const int16_t b = uart->read();
```

Comment:
better name for the variable?

Reply:
any variable name will probably match with an old typedef around the project, be free to give suggest names.